Content.Server/Disease/DiseaseZombieSystem.cs

```diff
@@ -0,0 +1,188 @@
+                {
+                    if (container.Value.ID == "gloves")
+                    {
+                        foreach (var entity in container.Value.ContainedEntities)
+                        {
+                            container.Value.Remove(entity);
+                        }
+                    }
+                }
+            }
+
+            if(TryComp<MindComponent>(uid, out var mindcomp))
+            {
+                if (mindcomp.Mind != null && mindcomp.Mind.TryGetSession(out var session))
+                {
+                    var chatMgr = IoCManager.Resolve<IChatManager>();
+                    chatMgr.DispatchServerMessage(session, Loc.GetString("zombie-infection-greeting"));
+                }
+            }
+
+            uid.PopupMessageEveryone(Loc.GetString("zombie-transform", ("target", uid)));
+            if (TryComp<MetaDataComponent>(uid, out var metacomp))
+            {
+                metacomp.EntityName = Loc.GetString("zombie-name-prefix", ("target", metacomp.EntityName));
+
+                if (!HasComp<GhostRoleMobSpawnerComponent>(uid)) //this specific component gives build test trouble so pop off, ig
+                {
+                    EntityManager.EnsureComponent<GhostTakeoverAvailableComponent>(uid, out var ghostcomp);
+                    ghostcomp.RoleName = metacomp.EntityName;
+                    ghostcomp.RoleDescription = Loc.GetString("zombie-role-desc"); 
+                    ghostcomp.RoleRules = Loc.GetString("zombie-role-rules");
+                }
+            }
+        }
+
+        /// <summary>
+        /// This handles zombie disease transfer when a entity is hit.
+        /// This is public because the virtualzombiehand entity needs it
+        /// </summary>
+        public void OnMeleeHit(EntityUid uid, DiseaseZombieComponent component, MeleeHitEvent args)
+        {
+            if (!args.HitEntities.Any())
+                return;
+
+            foreach (EntityUid entity in args.HitEntities)
+            {
+                if (entity == uid)
+                    continue;
+
+                if (!HasComp<MobStateComponent>(entity))
+                    continue;
+
+                if (_robustRandom.Prob(component.Probability)&& HasComp<DiseaseCarrierComponent>(entity))
+                {
+                    _disease.TryAddDisease(entity, "ZombieInfection");
+                }
+
+                EntityManager.EnsureComponent<MobStateComponent>(entity, out var mobState);
+                if ((mobState.IsDead() || mobState.IsCritical()) && !HasComp<DiseaseZombieComponent>(entity)) //dead entities are eautomatically infected. MAYBE: have activated infect ability?
+                {
+                    EntityManager.AddComponent<DiseaseZombieComponent>(entity);
+					var dspec = new DamageSpecifier();
+                    dspec.DamageDict.TryAdd("Slash", -12);
+                    dspec.DamageDict.TryAdd("Piercing", -7);
+					args.BonusDamage += dspec;
```

Comment:
what exactly is this trying to do? makes it not deal damage if it infects them? if so, just set the damage specifier to zero or something rather than hardcoding this